backend/app/users/routers.py

```diff
@@ -0,0 +1,87 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+from ..auth.dependencies import authenticate_either
+from ..database import get_async_session
+from .models import Users
+from .schemas import UserCreate, UserOut
+from .utils import hash_password
+
+TAG_METADATA = {
+    "name": "User Management",
+    "description": "Endpoints for adding, deleting and modifying users.",
+}
+
+router = APIRouter(
+    dependencies=[Depends(authenticate_either)],
```

Comment:
Should only admins we allowed to create users?

Reply:
I think - for now - yes? Seems logical to me